app/controllers/api/events_controller.rb

```diff
@@ -0,0 +1,41 @@
+class API::EventsController < ApplicationController
+
+  skip_before_action  :verify_authenticity_token
+  before_filter :set_access_control_headers
+
+  def create
+
+    @registered_app = RegisteredApp.find_by(url: request.env['HTTP_ORIGIN'])
+    @event = Event.new(event_params)
+    @event.registered_app_id = @registered_app.id
```

Comment:
You could also take advantage of Active Record associations to do `@event.registered_app = @registered_app` (the ID is set automatically) OR `@regiesterd_app.events.build`.  See [Association Reference](http://guides.rubyonrails.org/association_basics.html#detailed-association-reference)